src/gantry_calibration/calibration.py

```diff
@@ -0,0 +1,71 @@
+"""The core math and functionality behind the calibration algorithm."""
+
+import sys
+from collections import namedtuple
+import numpy as np
+
+
+Point = namedtuple("Point", ["x", "y"])
+PointPair = namedtuple("PointPair", ["expected", "actual"])
+
+
+class Calibration:
+    """Contains the calibration matrix and coefficients."""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, x_order: int, y_order: int, points: list[list[PointPair]]) -> None:
+        """Instantiate the object with expected and actual points."""
+        self.x_order = x_order
+        self.y_order = y_order
+        self.points = points
+        point_array = np.array(points)
+        self.x_count = point_array.shape[0]
```

Comment:
x_count and y_count should not be calculated at the time of initialization. They should be read-only properties that perform the calculation dynamically when the property is called. This makes it more robust in case the other data changes.

Reply:
Refactored with your suggestions of using properties.